<?xml version="1.0" ?>

<launch>
    <arg name="robot_type" default="humanoid"/>
    <!-- visualization config -->
    <arg name="rviz"               default="true" />
    <arg name="description_name"   default="humanoid_description"/>
    <arg name="multiplot"          default="false"/>

    <!-- make the files into global parameters -->
    <!-- The task file for the mpc. -->
    <arg name="taskFile"          default="$(find humanoid_interface)/config/mpc/task.info"/>
    <!-- The reference related config file of the robot -->
    <arg name="referenceFile"     default="$(find humanoid_interface)/config/command/reference.info"/>
    <!-- The URDF model of the robot -->
    <arg name="urdfFile"          default="/home/zhanghj/software/human_ws/src/humanoid-control/adam_lite/urdf/adam_lite_copy.urdf"/>
    <!-- <arg name="urdfFile"          default="$(find humanoid_legged_description)/mjcf/humanoid_legged.urdf"/> -->
    <arg name="urdfFileOrigin"    default="/home/zhanghj/software/human_ws/src/humanoid-control/adam_lite/urdf/adam_lite_copy.urdf"/>

    <arg name="urdfFileTest"    default="$(find humanoid_legged_description)/urdf/humanoid_legged_test.urdf"/>
    <!-- The file defining gait definition -->
    <arg name="gaitCommandFile"   default="$(find humanoid_interface)/config/command/gait.info"/>

    <!-- rviz -->
    <!-- <group if="$(arg rviz)">
      <param name="$(arg description_name)" textfile="$(arg urdfFileOrigin)"/>
      <arg name="rvizconfig" default="$(find humanoid_dummy)/rviz/humanoid.rviz" />
      <node pkg="rviz" type="rviz" name="rviz" args="-d $(arg rvizconfig)" output="screen" />
    </group> -->

    <!-- make the files into global parameters -->
    <param name="taskFile"          value="$(arg taskFile)" />
    <param name="referenceFile"     value="$(arg referenceFile)" />
    <param name="urdfFile"          value="$(arg urdfFile)" />
    <param name="urdfFileOrigin"          value="$(arg urdfFileOrigin)" />
    <param name="gaitCommandFile"   value="$(arg gaitCommandFile)"/>
    <param name="$(arg description_name)" textfile="$(arg urdfFileOrigin)"/>

    <node pkg="humanoid_dummy" type="humanoid_gait_command" name="humanoid_gait_command"
          output="screen" launch-prefix="gnome-terminal --"/>

    <node pkg="humanoid_controllers" type="humanoid_target_trajectories_publisher" name="humanoid_target"
          output="screen"/>
    <node pkg="humanoid_controllers" type="normal_controller_node" name="normal_controller"
          output="screen"/>
    <!-- <node pkg="mujoco_sim" type="humanoid_sim.py" name="humanoid_sim"
          output="screen"/> -->
    <node pkg="mujoco_sim" type="pnd_sim.py" name="humanoid_sim"
          output="screen"/>
    <node pkg="mujoco_sim" type="teleop.py" name="teleop"
          output="screen" launch-prefix="gnome-terminal --"/>
</launch>
